ansible_rulebook/rule_set_runner.py

```diff
@@ -284,6 +286,29 @@ def _handle_action_completion(self, task):
             if not self.action_loop_task.done():
                 self.action_loop_task.cancel()
 
+    async def _run_action_with_lock(
+        self,
+        lock_name: str,
+        action: str,
+        metadata: Metadata,
+        control: Control,
+        action_args: dict,
+    ):
+        """
+        Acquire the lock for the specified resource, access the resource,
+        and release the lock.
+        """
+        async with self.locks[
+            lock_name
+        ]:  # Acquire lock for the specific resource
+            logger.debug(f"Acquired lock for resource: {lock_name}")
```

Comment:
I think here we should also indicate the action and rule, specially if we are going to support the same lock across multiple actions/rules, otherwise I don't see how this message could be useful.